### PR TITLE
Add Suggestion interactor

### LINKED ISSUE #3764: Refactor SuggestionProviderContainer
Depends on #3807 
Related issue #3816

`ISuggestionProviderContainer` should be made into an interactor that returns `IObservable<IReadOnlyCollection<Suggestion>>`

The interactor should get suggestions from the providers and merge them and order them based on the probabilites introduced in #3807.

### PR DESCRIPTION
**Pull request by arturs105**
## What's this?
This PR removes `SuggestionProviderContainer` and adds an interactor as a replacement. The interactor is currently plain/dumb, but this PR is just to add a foundation on which we can build more complex suggestion logic.

The certainty logic isn't included in this PR, since it is still in review.

### Relationships

Closes #3764 

## :squid: Permissions
🦑

### CODE REVIEW

File: Toggl.Foundation.MvvmCross/ViewModels/SuggestionsViewModel.cs [hunk 1]
```diff
@@ -71,13 +67,8 @@ public override async Task Initialize()
...
         private IObservable<Suggestion[]> getSuggestions()
-        {
-            return suggestionProviders
-                .Providers
-                .Select(provider => provider.GetSuggestions())
-                .Aggregate(Observable.Merge)
-                .ToArray();
-        }
+            => interactorFactory.GetSuggestions(3).Execute()
+                .Select(suggestions => suggestions.ToArray());
```
[the remark this answers is not in the document]
Comment (arturs105) [reply Toggl.Foundation.MvvmCross/ViewModels/SuggestionsViewModel.cs]: The view model exposes suggestions as an array and it used in other places as an array, i didn't want to change the view model's public interface.

File: Toggl.Foundation.Tests/Toggl.Foundation.Tests.csproj [hunk 2]
```diff
@@ -33,6 +33,7 @@
     <Folder Include="Interactors\Workspace\" />
     <Folder Include="Interactors\Sync\" />
     <Folder Include="Interactors\Tag\" />
+    <Folder Include="Interactors\Suggestions\" />
```
[the remark this answers is not in the document]
Comment (arturs105) [reply Toggl.Foundation.Tests/Toggl.Foundation.Tests.csproj]: Why?

File: Toggl.Foundation.Tests/Interactors/Suggestions/GetSuggestionsInteractorTests.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    public sealed class GetSuggestionsInteractorTests
+    {
+        public sealed class TheConstructor : BaseInteractorTests
+        {
+            [Theory, LogIfTooSlow]
+            [ConstructorData]
+            public void ThrowsIfAnyOfTheArgumentsIsNull(bool useDatabase, bool useTimeService)
+            {
+                Action createInstance = () => new GetSuggestionsInteractor(
+                    3,
```
Comment (sofokli1) [inline Toggl.Foundation.Tests/Interactors/Suggestions/GetSuggestionsInteractorTests.cs]: Shouldn't this also be tested? is it ok to ask for 0 suggestions? what about null?
